src/Order/Bulk.php

```diff
@@ -340,4 +344,72 @@ public function render_messages() {
 			}
 		}
 	}
+
+	/**
+	 * Add Crete Label button to the action buttons within the order list table.
+	 *
+	 * @param  array  $actions  Order actions.
+	 * @param  \WC_Order  $order  Current order object.
+	 *
+	 */
+	public function add_create_label_actions_button( array $actions, \WC_Order $order ) {
+		// Display the button for all orders that have a PostNL as a shipping method.
+		if ( ! $this->is_postnl_shipping_method( $order ) ) {
+			return $actions;
+		}
+
+		if ( $this->have_backend_data( $order ) ) {
+			$actions['postnl-label'] = array(
+				'url'    => $this->get_download_label_url( $order->get_id() ),
+				'name'   => esc_html__( 'PostNL Print Label', 'postnl-for-woocommerce' ),
+				'action' => 'postnl-action-download-label',
+			);
+		} else {
+			$actions['postnl-label'] = array(
+				'url'    => wp_nonce_url( admin_url( 'admin-ajax.php?action=postnl_create_label&order_id=' . $order->get_id() ), 'postnl_create_label' ),
+				'name'   => esc_html__( 'PostNL Create Label', 'postnl-for-woocommerce' ),
+				'action' => 'postnl-action-create-label',
+			);
+		}
+
+		return $actions;
+	}
+
+	/**
+	 * Process PostNL by action button.
+	 */
+	public function postnl_create_label_ajax() {
+		if ( current_user_can( 'edit_shop_orders' ) && isset( $_GET['order_id'] ) ) {
+			$order_id = absint( wp_unslash( $_GET['order_id'] ) );
+
+			$array_messages = array(
+				'user_id' => get_current_user_id(),
+			);
+
+			try {
+				$result        = $this->save_meta_value( $order_id, $_REQUEST );
+				$tracking_note = $this->get_tracking_note( $order_id );
+
+				if ( $this->settings->is_woocommerce_email_enabled() && ! empty( $tracking_note ) ) {
+					$order = wc_get_order( $order_id );
+					$order->add_order_note( $tracking_note, 1 );
+				}
+
+				$array_messages[] = array(
+					'message' => sprintf( esc_html__( '#%1$s : PostNL label has been created.', 'postnl-for-woocommerce' ), $order_id ),
+					'type'    => 'success',
+				);
+			} catch ( \Exception $e ) {
+				$array_messages[] = array(
+					'message' => sprintf( '#%1$s : %2$s', $order_id, $e->getMessage() ),
+					'type'    => 'error',
+				);
+			}
```

Comment:
It seems this code is redundant with [this code](https://github.com/Progressus-io/postnl-for-woocommerce/blob/d77024ab20addc5940dce4baf1a723fd50966009/src/Order/Bulk.php#L81-L108) under `process_order_bulk_actions()` method. Do you think we can make new method for this?

Reply:
@iyut Yes, that what I thought, I'm thinking of making a method that return array of messages & labels, what do you  think ? do you have better idea?

Reply:
> I'm thinking of making a method that return array of messages & labels
I think this is a good idea! Let's do that!